File: less3/e/main.go
package main

import (
	"io"
	"os"

	. "github.com/aaa2ppp/contestio"
)

var debug bool

const modulo = 1000000007

type solveFunc func([]uint64) uint64

func solve(a []uint64) uint64 {
	n := len(a)
	b := make([]uint64, n+1)
	for i, v := range a {
		b[i+1] = (b[i] + v) % modulo
	}

	var ans uint64
	for i := 1; i < n-1; i++ {
		v := a[i]
		v = (v * (b[i] + (modulo - b[0]))) % modulo
		v = (v * (b[n] + (modulo - b[i+1]))) % modulo
		ans = (ans + v) % modulo
	}
	return ans
}

func run(in io.Reader, out io.Writer, solve solveFunc) {
	br := NewReader(in)
	bw := NewWriter(out)
	defer bw.Flush()

	var n int
	var a []uint64

	ScanIntLn(br, &n)

	a = Resize(a, n)
	ScanInts(br, a)

	ans := solve(a)
	PrintIntLn(bw, ans)
}

func main() {
	run(os.Stdin, os.Stdout, solve)
}

File: less3/e/main.py
import sys

MOD = 1_000_000_007

def solve(a):
    n = len(a)
    if n < 3:
        return 0

    b = [0] * (n + 1)
    for i, val in enumerate(a):
        b[i + 1] = (b[i] + val) % MOD

    ans = 0
    for i in range(1, n - 1):
        v = a[i]
        v = (v * b[i]) % MOD
        v = (v * ((b[n] - b[i + 1]))) % MOD
        ans = (ans + v) % MOD

    return ans


def main():
    n = int(input())
    a = list(map(int, input().split()))
    ans = solve(a)
    print(ans)


if __name__ == "__main__":
    main()
